internal/component/database_observability/mysql/collector/parser/parser.go

```diff
@@ -0,0 +1,23 @@
+package parser
```

Comment:
Thanks for the fixes @dehaansa! I do have a slight preference for leaving the interface declaration close to the caller, but it's not a big deal.

Reply:
Understood - this change was to avoid a cyclical dependency after adding the StatementType type.